_posts/2020-10-02-nlp-translation.md

```diff
@@ -0,0 +1,205 @@
+updating weights.
+
+### Datasets
+Both tasks use the same test set, but are trained on slightly different data sets:
+- The LSTM is trained on the English to German World Machine Translation 16 (WMT16) dataset, comprising of 3,975,116 translated sentences
+- The Transformer is trained on the English to German World Machine Translation 17 (WMT17) dataset, comprising of 4,590,101.
+ 
+
+More details on both tasks can be found on our [documentation](https://mlbench.readthedocs.io/en/latest/benchmark-tasks.html#task-4-machine-translation).
+
+## Results
+
+Let us now get to fun part; the results. As previously discussed, those models have important training times, and the aim of MLBench is to study the benefit of distribution.
+For reproducibility purposes, here is the hardware and software we have used:
+- Cloud service: Google Cloud
+- Machine Type: `n1-standard-4`
+- PyTorch 1.5.1
+- NVIDIA Tesla-T4 GPU (1 per node)
+- 4 cores and 15GB of RAM
+- NCCL communication backend 
+
+The goal for both models is determined by the Bilingual Evaluation Understudy Score (BLEU):
+- The LSTM task stops when reaching a BLEU Score of 24.0
+- The Transformer task stops when reaching a BLEU score of 25.0
+
+The models are trained on 1,2,4,8 and 16 workers, and all step times are precisely measured to obtain an accurate speed up quantification.
+Speedups are computed with respect to the 1 worker case, and are intended to illustrate the distributive capabilities of the task.
+
+### Overall Speedups
+
+The graphs below show the time speedups for the LSTM model and Transformer model (respectively). 
+
+<a href="{{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_speedup.png" data-lightbox="task4a_speedups" data-title="Speedups for GNMT">
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_speedup.png)
+  *GNMT Speedups*
+</a>
+
+<a href="{{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4b_speedup.png" data-lightbox="task4b_speedups" data-title="Speedups for Transformer">
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4b_speedup.png)
+  *Transformer Speedups*
+</a>
+
+The left graph shows the absolute speed ups with respect to one worker, and the right one omits
+communication times from the speed up. This allows us to better see the effect of communication.
+
+
+A few interesting points:
+- Overall speedups follows a $$ log_{2}(n) $$, with `n = num_workers`, while compute are roughly linear.
+- Scaling the number of compute nodes gives nearly perfect scaling for both tasks (right plot)
+- Using more powerful communication hardware (e.g. Tesla V100) will positively affect speedups.
+
+As the distribution level increases, we can see that communication becomes more and more heavy, and attenuates speedups quite significantly.
+
+### Step times
+
+The next figures show the total time spent in each step of training. 
+
```

Comment:
It's not completely clear to me which graphs 'include' which others. Maybe you could add some kind of diagram that shows something like
```
computation = forward + backward
total = computation + communication
```
for each of the labels used.